airflow/models.py

```diff
@@ -1795,6 +1795,9 @@ def get_template_context(self, session=None):
         if task.params:
             params.update(task.params)
 
+        if dag_run and dag_run.conf:
+            params.update(dag_run.conf)
```

Comment:
I'm not sure about this. `params` is documented and has a clear hierarchy, and nowhere it says that your DAGRun conf will override what is in there. I don't think of the dag_run's configuration as something in the same namespace as `params`. Even if that was the case, should the conf be higher in the override hierarchy? It surely is neither intuitive or documented clearly here.

This is not backward compatible and could introduce rare and intricate collisions/change of behavior.

Plus DagRun.conf is already available in the template through `{{ dag_run.conf }}`, why is there a need to merge it into `params`

Reply:
@mistercrunch We want to pass some key-value pairs through `backfill -p '{"text": "blah blah"}'`or `trigger_dag -p '{"text": "blah blah"}'` to override the value in {{ params.text }}. 
Do you have recommended ways to do that?